tests/phpunit/Civi/Funding/FundingCase/Api4/ActionHandler/SetRecipientContactActionHandlerTest.php

```diff
@@ -0,0 +1,125 @@
+use Civi\Funding\EntityFactory\FundingCaseFactory;
+use Civi\Funding\EntityFactory\FundingCaseTypeFactory;
+use Civi\Funding\EntityFactory\FundingProgramFactory;
+use Civi\Funding\FundingCase\Command\FundingCaseRecipientContactSetCommand;
+use Civi\Funding\FundingCase\FundingCaseManager;
+use Civi\Funding\FundingCase\Handler\FundingCaseRecipientContactSetHandlerInterface;
+use Civi\Funding\FundingProgram\FundingCaseTypeManager;
+use Civi\Funding\FundingProgram\FundingProgramManager;
+use Civi\Funding\Traits\CreateMockTrait;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Civi\Funding\FundingCase\Api4\ActionHandler\SetRecipientContactActionHandler
+ */
+final class SetRecipientContactActionHandlerTest extends TestCase {
+
+  use CreateMockTrait;
+
+  /**
+   * @var \Civi\Funding\ApplicationProcess\ApplicationProcessManager&\PHPUnit\Framework\MockObject\MockObject
+   */
+  private MockObject $applicationProcessManagerMock;
+
+  private SetRecipientContactActionHandler $actionHandler;
+
+  /**
+   * @var \Civi\Funding\FundingCase\FundingCaseManager&\PHPUnit\Framework\MockObject\MockObject
+   */
+  private MockObject $fundingCaseManagerMock;
+
+  /**
+   * @var \Civi\Funding\FundingProgram\FundingCaseTypeManager&\PHPUnit\Framework\MockObject\MockObject
+   */
+  private MockObject $fundingCaseTypeManagerMock;
+
+  /**
+   * @var \Civi\Funding\FundingProgram\FundingProgramManager&\PHPUnit\Framework\MockObject\MockObject
+   */
+  private MockObject $fundingProgramManagerMock;
+
+  /**
+   * @var \Civi\Funding\FundingCase\Handler\FundingCaseRecipientContactSetHandlerInterface&\PHPUnit\Framework\MockObject\MockObject
+   */
+  private MockObject $recipientContactSetHandlerMock;
+
+  protected function setUp(): void {
+    parent::setUp();
+    $this->applicationProcessManagerMock = $this->createMock(ApplicationProcessManager::class);
+    $this->recipientContactSetHandlerMock = $this->createMock(FundingCaseRecipientContactSetHandlerInterface::class);
+    $this->fundingCaseManagerMock = $this->createMock(FundingCaseManager::class);
+    $this->fundingCaseTypeManagerMock = $this->createMock(FundingCaseTypeManager::class);
+    $this->fundingProgramManagerMock = $this->createMock(FundingProgramManager::class);
+
+    $this->actionHandler = new SetRecipientContactActionHandler(
+      $this->applicationProcessManagerMock,
+      $this->fundingCaseManagerMock,
+      $this->fundingCaseTypeManagerMock,
+      $this->fundingProgramManagerMock,
+      $this->recipientContactSetHandlerMock
+    );
+  }
+
+  public function testUpdateAmountApproved(): void {
+    $action = $this->createApi4ActionMock(SetRecipientContactAction::class);
+    $action->setId(FundingCaseFactory::DEFAULT_ID)
+      ->setContactId(1234);
+
+    $fundingCase = FundingCaseFactory::createFundingCase();
+    $this->fundingCaseManagerMock->method('get')
+      ->with($fundingCase->getId())
+      ->willReturn($fundingCase);
```

Comment:
for discussion: 
maybe one could define a trait with data for the mocks, such as _funding-case id, type-id, programm-id_ instead of taking it from the `$fundingCase` object? currently any (accidential) change of the data in the factory method (ie. setting a value to an empty string or so) that creates a `$fundingCase` will sneak into this test, maybe without being noticed by the test.

having a separate set of test data for the mocks would ensure that a change of values used by the funding case factory would most likely be noticed by failing tests, as well as the opposite, a change in the test data for the mocks.